baseplate/integration/pyramid.py

```diff
@@ -71,26 +91,23 @@ def _on_new_request(self, event):
             except (KeyError, ValueError):
                 pass
 
-        request.start_root_span(request.matched_route.name, trace_info)
+        request.start_server_span(request.matched_route.name, trace_info)
+        request.trace.set_tag("http.url", request.url)
+        request.trace.set_tag("http.method", request.method)
+        request.trace.set_tag("peer.ipv4", request.remote_addr)
 
-    def _start_root_span(self, request, name, trace_info=None):
-        request.trace = self.baseplate.make_root_span(
+    def _start_server_span(self, request, name, trace_info=None):
+        request.trace = self.baseplate.make_server_span(
             request,
             name=name,
             trace_info=trace_info,
         )
         request.trace.start()
 
-    # pylint: disable=no-self-use
-    def _on_new_response(self, event):
-        if not event.request.matched_route:
-            return
-
-        event.request.trace.stop()
-
     def includeme(self, config):
-        config.add_subscriber(self._on_new_request, ContextFound)
-        config.add_subscriber(self._on_new_response, NewResponse)
+        config.add_subscriber(self._on_new_request, pyramid.events.ContextFound)
+        config.add_tween("baseplate.integration.pyramid._make_baseplate_tween",
+                         under=pyramid.tweens.EXCVIEW)
```

Comment:
🔕 
Purely educational question. I'm curious about when you use these different but similar parts of pyramid.  It seems like you want to use a tween here because you want to respond to something but not change it.  As you said in the commit message, a NewResponse subscription was too late in the flow to do what you wanted.  It also seems like you _could_ use add_view with context=Exception but you don't because I'm guessing it would mess up if a baseplate service (like the gateway) added an Exception view as well and adding a view is for when you actually want to change the response and in this case, you just want this to be a side effect.